lib/puppet/indirector/catalog/rest.rb

```diff
@@ -20,6 +20,7 @@ def find(request)
       facts: request.options[:facts_for_catalog],
       environment: request.environment.to_s,
       configured_environment: request.options[:configured_environment],
+      check_environment: request.options[:check_environment],
```

Comment:
Seems not necessary to use double bang here

Reply:
nvm, from the commit, "The double negation is needed in case `nil` is passed and behaves the same as `false`."